contracts/NewForwarder.sol

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.15;
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Receiver.sol';
+import './ERC20Interface.sol';
+import './TransferHelper.sol';
+import './IForwarder.sol';
+
+/**
+ * Contract that will forward any incoming Ether to the parent address of the contract
+ *
+ */
+contract NewForwarder is IERC721Receiver, ERC1155Receiver, IForwarder {
+  // Address to which any funds sent to this contract will be forwarded
+  address public parentAddress;
+  address public feeAddress;
+  mapping(address => bool) public whiteListedAddresses;
+  bool public autoFlush721 = true;
+  bool public autoFlush1155 = true;
+
+  event ForwarderDeposited(address from, uint256 value, bytes data);
+
+  /**
+   * Initialize the contract, and sets the destination address to that of the parent address
+   */
+  function init (
+    address _parentAddress,
+    address _feeAddress,
+    bool _autoFlush721,
+    bool _autoFlush1155
+  ) external onlyUninitialized {
+    parentAddress = _parentAddress;
+    feeAddress = _feeAddress;
```

Comment:
Should we add null address checks here in constructor

Reply:
updated